analytics_export/analyticsExport.js

```diff
@@ -1,8 +1,197 @@
+
+    const dimensionGiven = dimension !== undefined && dimension !== null;
+
+    const measures = measure instanceof Array ? measure : [measure];
+    const queryConfig = {
+      measures,
+      frequency: itc.frequency.days,
+    };
+    if (dimensionGiven) {
+      queryConfig.group = { dimension };
+    }
+    const query = itc.AnalyticsQuery.metrics(this.appId, queryConfig).date(startDate, endDate);
+
+    // Directly fetch instead of itc.request to avoid queue and properly handle errors
+    const response = await fetch(
+      url.parse(query.apiURL + query.endpoint),
+      {
+        method: 'POST',
+        body: JSON.stringify(query.assembleBody()),
+        headers: this.client.getHeaders(),
+      },
+    );
+    const data = await response.json();
+    if (response.ok === false) {
+      throw new RequestError(
+        `${response.status} ${response.statusText}
+        \n${JSON.stringify(data.errors, null, 2) || ''}`,
+        response.status,
+      );
+    }
+
+    const resultsByDate = new Map();
+
+    data.results
+      .filter((result) => result.totals.value !== -1)
+      .flatMap((result) => (
+        result.data.map((dayData) => {
+          const value = {
+            date: dayData.date.slice(0, 10),
+            app_name: this.appName,
+            value: dayData[measure],
+          };
+          if (dimensionGiven) {
+            value[dimension] = result.group.title;
+          }
+          return value;
+        })
+      )).forEach((value) => {
+        if (!resultsByDate.has(value.date)) {
+          resultsByDate.set(value.date, []);
+        }
+        resultsByDate.get(value.date).push(value);
+      });
+
+    return resultsByDate;
+  }
+
+  /**
+   * Pause thread for number of seconds; can be used for exponential backoff in requests
+   */
+  static async sleep(seconds) {
+    return new Promise(((resolve) => setTimeout(resolve, seconds * 1000)));
+  }
+
+  async startExport(startDate, endDate, overwrite) {
+    const parsedStartDate = Date.parse(startDate);
+    const parsedEndDate = Date.parse(endDate);
+    if (Number.isNaN(parsedStartDate) || Number.isNaN(parsedEndDate)) {
+      throw Error('Execution dates must be given in the format YYYY-MM-DD');
+    } else if (parsedStartDate > parsedEndDate) {
+      throw Error('Start date must be before end date');
+    }
+
+    console.log('Starting export');
+
+    const measuresByDimension = await this.getAllowedDimensionsPerMeasure(
+      parsedStartDate, parsedEndDate,
+    ).catch((err) => { throw new Error(`Failed to get analytics metadata: ${err}`); });
+
+    const bqClient = await BigqueryClient.createClient(this.project, this.dataset)
+      .catch((err) => { throw Error(`Failed to create bigquery client: ${err}`); });
+
+    // Run in for loop to synchronously send each request to avoid hitting rate limit
+    for (const [dimension, measures] of measuresByDimension) {
+      if (!(dimension in dimensionToTableSuffix)) {
```

Comment:
```suggestion
      if (!dimensionToTableSuffix.includes(dimension)) {
```

Reply:
I think includes is a lodash thing.  Doesn't look like it works here

Reply:
hmm, it should be part of the ES API (you know something is lodash because you need to use `_.foo`) but maybe the version of node you're using is too old